waspc/cli/Command/CreateNewProject.hs

```diff
@@ -0,0 +1,68 @@
+module Command.CreateNewProject
+    ( createNewProject
+    ) where
+
+import System.Directory (createDirectory, getCurrentDirectory)
+import qualified System.FilePath as FP
+import qualified Path as P
+import Text.Printf (printf)
+import Control.Monad.IO.Class (liftIO)
+
+import StrongPath (Path, Rel, File, (</>))
+import qualified StrongPath as SP
+import ExternalCode (SourceExternalCodeDir)
+import Command (Command)
+import qualified Common
+
+
+createNewProject :: String -> Command ()
+createNewProject projectName = do
+    absCwd <- liftIO getCurrentDirectory
+    waspProjectDir <- SP.parseAbsDir $ absCwd FP.</> projectName
+    liftIO $ do
+        createDirectorySP waspProjectDir
+        writeFileSP (waspProjectDir </> mainWaspFileInWaspProjectDir) mainWaspFileContent
+        writeFileSP (waspProjectDir </> gitignoreFileInWaspProjectDir) gitignoreFileContent
+        writeFileSP (waspProjectDir </> Common.dotWaspRootFileInWaspProjectDir)
+            "File marking the root of Wasp project."
+
+    let extCodeDir = waspProjectDir </> Common.extCodeDirInWaspProjectDir
+    liftIO $ do
+        createDirectorySP extCodeDir
+        writeFileSP (extCodeDir </> mainPageJsFileInExtCodeDir) mainPageJsFileContent
+
+    liftIO $ putStrLn $ "Created new Wasp project in ./" ++ projectName ++ " directory!"
+  where
+      mainWaspFileInWaspProjectDir :: Path (Rel Common.WaspProjectDir) File
+      mainWaspFileInWaspProjectDir = SP.fromPathRelFile [P.relfile|main.wasp|]
+      mainWaspFileContent = unlines
```

Comment:
this could be in some template, right? So it's not hardcoded like this. but ok

Reply:
wasp generating itself, nice :D

Reply:
It could but that was more complicated for now, I just wanted to get us running. Yes cool :D.